metadeploy/api/models.py

```diff
@@ -847,6 +862,56 @@ def run(self, ctx, plan, steps, org):
         flow_coordinator.run(org)
 
 
+class ScratchOrg(HashIdMixin, models.Model):
+    Status = Choices("started", "complete", "failed", "canceled")
+
+    plan = models.ForeignKey(Plan, on_delete=models.CASCADE)
+    email = models.EmailField()
+
+    enqueued_at = models.DateTimeField(null=True)
+    job_id = models.UUIDField(null=True)
+    # This is set in a user's session to let them continue to access
+    # this job, without being otherwise auth'd:
+    uuid = models.UUIDField(null=True)
+    created_at = models.DateTimeField(auto_now_add=True)
+    edited_at = models.DateTimeField(auto_now=True)
+    status = models.CharField(choices=Status, max_length=64, default=Status.started)
+    config = JSONField(null=True, blank=True, encoder=DjangoJSONEncoder)
+    org_id = models.CharField(null=True, blank=True, max_length=18)
+
+    def save(self, *args, **kwargs):
+        ret = super().save(*args, **kwargs)
+        if not self.enqueued_at:
+            from .jobs import create_scratch_org_job
+
+            job = create_scratch_org_job.delay(
+                plan_id=str(self.plan.id),
+                email=self.email,
```

Comment:
Let's avoid storing the email in redis. create_scratch_org_job can read it from the ScratchOrg record identified by result_id.